# Skyline Enrollment: Confidence Intervals

Lesson 1.5 Practice Excercise. Computes confidence intervals for means and
proportions on the skyline enrollment synthetic dataset

Author: Meron Welderufael
Date: 08/14/2026

### CI for a mean -  95% confidence interval for the mean of hours_studied

In [1]:

import numpy as np
import pandas as pd
from scipy.stats import stats


In [6]:
skyline_enrollment = pd.read_csv("C:/Users/Lenovo/northstar-coursework/module-01-foundations-of-analytics-and-statistics/lesson-1-2-types-of-data/skyline_enrollments.csv")

n_students = len(skyline_enrollment)
sample_mean = skyline_enrollment["hours_studied"].mean()
sample_std = skyline_enrollment["hours_studied"].std()

#standard error of the mean
standard_error = sample_std / np.sqrt(n_students)

#critical value for 95% confidence level
z= 1.96

#margin of error
margin_of_error = z * standard_error

#confidence_interval 

ci_lower = sample_mean - margin_of_error
ci_upper = sample_mean + margin_of_error

print(f"sample size = {n_students}")
print(f"sample mean = {sample_mean:.2f}")
print(f"sample standard deviation = {sample_std:.2f}")

print(f"standard error = {standard_error:.2f}")
print(f"margin of error = {margin_of_error:.2f}")
print(f"95% confidence interval = ({ci_lower:.2f}, {ci_upper:.2f})")

sample size = 100
sample mean = 30.14
sample standard deviation = 3.36
standard error = 0.34
margin of error = 0.66
95% confidence interval = (29.48, 30.80)


As we can see, the 95% confidence interval is between 29.48 and 30.8. If we repeated the sampling process many times, about 95% of the resulting intervals would contain the true mean of the student population.

### CI for a proportion

In [14]:
n_students = len(skyline_enrollment)
n_completed = (skyline_enrollment["completion_status"] == "Completed").sum()

sample_prop = n_completed / n_students

# standard error of the proportion
se_prop = np.sqrt((sample_prop * (1 - sample_prop)) / n_students)

#95% CI
z= 1.96
moe_prop = z * se_prop
ci_lower = sample_prop - moe_prop
ci_upper = sample_prop + moe_prop

print(f"sample size = {n_students}")
print(f"Completed status from sample = {sample_prop:.2f}")
print(f"standard error of proportion = {se_prop:.2f}")
print(f"margin of error proportion = {moe_prop:.2f}")
print(f"95% CI completed status = ({ci_lower:.2f}, {ci_upper:.2f})")



sample size = 100
Completed status from sample = 0.82
standard error of proportion = 0.04
margin of error proportion = 0.08
95% CI completed status = (0.74, 0.90)


The sample completion rate is 82%, with a 95% confidence interval ranging from 74% to 90%. This suggests that the true completion rate for the student population is likely around this range.

The current margin of error is about ±8 percentage points with a sample size of 100. To reduce the margin of error to ±2 percentage points, the sample size would need to increase by about (8/2)= 4,  4^2= 16 times. Therefore, Skyline would need a sample of roughly 100*16 = 1,600 students.